imports/components/controls.jsx

```diff
@@ -0,0 +1,204 @@
+import { DropdownButton, MenuItem, Modal, Button } from 'react-bootstrap';
+import { PlayerList } from './player_list';
+
+class RevealControl extends React.Component {
+  render() {
+    return (
+      <DropdownButton title="Reveal" id='dReveal' onSelect={this.props.doReveal}>
+        <MenuItem data-target='square'>Square</MenuItem>
+        <MenuItem data-target='word'>Word</MenuItem>
+        <MenuItem data-target='grid'>Grid</MenuItem>
+      </DropdownButton>
+    )
+  }
+}
+
+class CheckControl extends React.Component {
+  render() {
+    return (
+      <DropdownButton className={this.props.checkOk && "check-ok"} title="Check" id='dCheck' onSelect={this.props.doCheck}>
+        <MenuItem data-target='square'>Square</MenuItem>
+        <MenuItem data-target='word'>Word</MenuItem>
+        <MenuItem data-target='grid'>Grid</MenuItem>
+      </DropdownButton>
+    )
+  }
+}
+
+class PencilControl extends React.Component {
+  render() {
+    return (
+      <div className="btn-group">
+        <button data-pencil="false"
+                className={'btn' + (this.props.isPencil ? '' : ' active')}
```

Comment:
Maybe worth it, maybe not, but I used https://www.npmjs.com/package/classnames for this type of logic in my other project and was decently happy with it